src/Domain/Model/MembershipApplication.php

```diff
@@ -30,24 +32,27 @@ class MembershipApplication {
 	private $needsModeration;
 	private $isCancelled;
 
-	public static function newApplication( string $type, MembershipApplicant $applicant, MembershipPayment $payment ): self {
+	public static function newApplication( string $type, MembershipApplicant $applicant, MembershipPayment $payment,
+										   MembershipApplicationTrackingInfo $trackingInfo ): self {
 		return new self(
 			null,
 			$type,
 			$applicant,
 			$payment,
+			$trackingInfo,
 			self::NO_MODERATION_NEEDED,
 			self::IS_CURRENT
 		);
 	}
 
 	public function __construct( int $id = null, string $type, MembershipApplicant $applicant, MembershipPayment $payment,
-		bool $needsModeration, bool $isCancelled ) {
+		MembershipApplicationTrackingInfo $trackingInfo, bool $needsModeration, bool $isCancelled ) {
 
 		$this->id = $id;
 		$this->type = $type;
 		$this->applicant = $applicant;
 		$this->payment = $payment;
+		$this->trackingInfo = $trackingInfo;
```

Comment:
Err, right, the `MembershipApplicationTracker` interface. I must have been daydreaming when I merged #417... While implementing the tracking data persistence, I wondered why we want a service for that. Just like any other data the tracking stuff comes as a request parameter and can be handled as such.

Maybe I am missing something here, what's the advantage in having this handled by a service?